cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -212,7 +212,18 @@ if (CMAKE_BUILD_TYPE MATCHES "Debug")
   #
   # https://github.com/dealii/dealii/issues/15496
   #
-  enable_if_supported(DEAL_II_CXX_FLAGS_DEBUG "-ffp-exception-behavior=strict")
+  # ... except this doesn't presently work with AppleClang versions before 16.0.
+  # They support this flag but it is not compatible with C++14: see
+  #
+  # https://github.com/dealii/dealii/issues/15531
+  #
+  # This flag works with standard Clang 16.0 or newer.
+  #
+  if((CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
+        OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+      AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "16.0")
+    enable_if_supported(DEAL_II_CXX_FLAGS_DEBUG "-ffp-exception-behavior=strict")
+  endif()
```

Comment:
I thought we only run into problems with `AppleClang`. Is there a good reason to also disable the check for regular `clang` and restrict to version `>=16.0`? AFAICT, the flag is supported from `clang-10` on.

Reply:
I set it to 16 since I'm sure that version works (it is what we used in the original PR). Do you have access to clang-10? If so, can you verify that that flag works with C++14?

Reply:
https://godbolt.org/z/o7Eb1YPac shows that `clang-10` doesn't warn when using the flag but I haven't tried compiling `deal.II` with it. I would be curious why `enable_if_supported` doesn't quite work here.

Reply:
I decreased the minimum clang version to 13 (@tamiko told me that these are all known to work)